### ISSUE TITLE
Multilabel classification training data

### ISSUE THREAD

**Issue by josh-yang92** [huggingface/setfit#413]
Hi guys, just have a question regarding the training data for multilabel classification.

So, for multiclass classification, you can play around with the number of samples (K) per label, and of course, the higher the K, the performance increases; this is straightforward as there will be only 1 label per sample.

However, for multilabel classification, where there can be more than 1 label per sample (also in many different combinations), how are we supposed to construct our training data? for example, would it be best to give an equal number of samples for every combination of labels? this would exponentially increase the training data required which would defeat the purpose of few-shot learning...?

I am asking this question as I have tried training the model without considering the above question and getting not-so-great results (around 65% accuracy with 12 labels).

Thank you!

**Comment by MattiL** [comment]
I have used SetFit when there is a limited number of labeled training data available. If you have ample data, you can also use other frameworks. I have used the following code to balance unbalanced training data classes.
```
model = SetFitModel(
                    model_body=SentenceTransformer('all-MiniLM-L6-v2'), 
                    model_head=OneVsRestClassifier(LogisticRegression(class_weight="balanced")),
                    multi_target_strategy="one-vs-rest"
                    )
```
[](https://github.com/huggingface/setfit/issues/375#issuecomment-1622412726)

**Comment by josh-yang92** [comment]
@MattiL I am not sure if you understood my question correctly. I am talking about multilabel problem where the input data can have multiple labels at the same time out of many labels unlike multiclass problem where the data can have exactly one label out of many.

To balance the training data for the multiclass problem, it's easy, you just balance the data or adopt the method like you have. However, for the multilabel problem, since you can have sum(nCr) combinations, you w

**Comment by MattiL** [comment]
I have tried to use the balancing code for multilabel classification. I guess that might improve the accuracy. Multiclass has had little support in SetFit.

**Comment by alejandrodumas** [comment]
You could use [scikit-multilearn](http://scikit.ml/#) to create a balanced training dataset. Use `iterative_train_test_split`

**Comment by singularity014** [comment]
to all wondering how the data should look like. Here is a sample format

`text,vocab,slang,idioms,simple
"Exhausted from a relentless schedule, the formidable entrepreneur persevered through arduous obstacles to achieve unprecedented success.",1,0,0,0
"After witnessing the catastrophic impact of deforestation firsthand, the environmentalist fervently advocated for reforestation measures to mitigate irreversible ecological damage.",1,0,0,0
The arduous task of persuading the obstinate toddler to eat his vegetables became an exhausting battle of wills.,1,0,0,0
"Amidst unprecedented global turmoil, the populace strives relentlessly to find solace in the pursuit of self-actualization.",1,0,0,0
The arduous pursuit of success often leads individuals to forsake their personal relationships in favor of their professional ambitions.,1,0,0,0
"Despite encountering numerous obstacles throughout his journey, the resilient entrepreneur persistently strategized and ultimately achieved remarkable success in his business venture.",1,0,0,0`
